src/ngx_srt_set_misc_module.c

```diff
@@ -0,0 +1,302 @@
+      ngx_srt_set_misc_decrypt,
+      NGX_SRT_MAIN_CONF_OFFSET,
+      0,
+      NULL },
+
+      ngx_null_command
+};
+
+static ngx_srt_module_t  ngx_srt_set_misc_module_ctx = {
+    NULL,                                  /* preconfiguration */
+    NULL,                                  /* postconfiguration */
+
+    NULL,                                  /* create main configuration */
+    NULL,                                  /* init main configuration */
+
+    NULL,                                  /* create server configuration */
+    NULL                                   /* merge server configuration */
+};
+
+ngx_module_t  ngx_srt_set_misc_module = {
+    NGX_MODULE_V1,
+    &ngx_srt_set_misc_module_ctx,             /* module context */
+    ngx_srt_set_misc_commands,                /* module directives */
+    NGX_SRT_MODULE,                        /* module type */
+    NULL,                                  /* init master */
+    NULL,                                  /* init module */
+    NULL,                                  /* init process */
+    NULL,                                  /* init thread */
+    NULL,                                  /* exit thread */
+    NULL,                                  /* exit process */
+    NULL,                                  /* exit master */
+    NGX_MODULE_V1_PADDING
+};
+
+
+static ngx_int_t
+ngx_srt_set_misc_base64_variable(ngx_srt_session_t *s, ngx_srt_variable_value_t *v,
+    uintptr_t data)
+{
+    ngx_str_t              val, decode_str;
+    ngx_srt_base64_ctx_t  *base64;
+
+    base64 = (ngx_srt_base64_ctx_t *) data;
+
+    ngx_log_debug0(NGX_LOG_DEBUG_STREAM, s->connection->log, 0,
+        "srt base64 started");
+
+    if (ngx_srt_complex_value(s, &base64->value, &val) != NGX_OK) {
+        ngx_log_error(NGX_LOG_ERR, s->connection->log, 0,
+        "ngx_srt_set_misc_base64_variable: failed to eval complex value");
+        return NGX_ERROR;
+    }
+
+    decode_str.len = ngx_base64_decoded_length(val.len);
+    decode_str.data = ngx_pnalloc(s->connection->pool, decode_str.len);
+    ngx_decode_base64(&decode_str, &val);
+    v->valid = 1;
+    v->no_cacheable = 0;
+    v->not_found = 0;
+    v->len = decode_str.len;
+    v->data = decode_str.data;
+
+    ngx_log_debug2(NGX_LOG_DEBUG_STREAM, s->connection->log, 0,
+                    "srt base64: \"%V\" \"%v\"", &val, v);
+
+    return NGX_OK;
+}
+
+static char *
+ngx_srt_set_misc_base64(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
+{
+    ngx_str_t                       *value, name;
+    ngx_srt_variable_t              *var;
+    ngx_srt_base64_ctx_t            *base64;
+    ngx_srt_compile_complex_value_t  ccv;
+
+    base64 = ngx_pcalloc(cf->pool, sizeof(ngx_srt_base64_ctx_t));
+    if (base64 == NULL) {
+        return NGX_CONF_ERROR;
+    }
+
+    value = cf->args->elts;
+
+    ngx_memzero(&ccv, sizeof(ngx_srt_compile_complex_value_t));
+
+    ccv.cf = cf;
+    ccv.value = &value[1];
```

Comment:
better have set_decode_base64 {dst} {src} (reverse the order of the args), because that's the convention in all other set directives